### ISSUE TITLE
[BUG] MissingMethodException when project build and deployed via Visual Studio 2022

### ISSUE THREAD

**Issue by mgramley** [reactiveui/Akavache#735]
Attempting to insert an object into the `BlobCache` via `BlobCache.LocalMachine.InsertObject` fails on Xamarin.Forms (Android) when **my project is built using Visual Studio 2022**. 

It throws a `MissingMethodException`:
```
System.MissingMethodException: Method not found: int SQLitePCL.raw.sqlite3_bind_blob(SQLitePCL.sqlite3_stmt,int,System.ReadOnlySpan`1<byte>)
  at Akavache.Sqlite3.SqliteOperationQueue.MarshalCompletion (System.Object completion, System.Action block, System.IObservable`1[T] commitResult) [0x00007] in D:\a\Akavache\Akavache\src\Akavache.Sqlite3\Queues\OperationQueue.cs:395 
```

Note that this same code works when **built using Visual Studio 2019**. I don't believe this is a linker issue as I have turned off that setting and this still persists. 

Has anyone else seen this issue? Are there any workarounds? Obviously, I can keep working in 2019, but not being able to use the most recent IDE's is a pain.

**Comment by glennawatson** [comment]
Might be worth making a sample if you can. 

Other thing that can commonly cause this is a version mismatch with the nuget package

**Comment by mgramley** [comment]
So I'll try to create a sample project with this when I can. 

Which NuGet packages would be mismatched? I think I have the latest installed, unless you mean a mismatch with one of its dependencies?

**Comment by glennawatson** [comment]
Thinking the sqlite package.

**Comment by mgramley** [comment]
So I don't reference the sqlite package directly, I assume it just gets resolved since it's an Akavache dependency. I checked my other dependencies to see if any relied on sqlite, but they don't, so I'm not sure what it would be mismatching with.

Again, this works just fine when I build and deploy from VS 2019. It's only causing this issue when deployed from VS 2022. I would assume that a package mismatch issue would show up for both.

**Comment by protokovich** [comment]
same problem on `await BlobCache.Secure.GetObjects<T>(keys)`

`System.MissingMemberException`

`Method not found: System.ReadOnlySpan1<byte> SQLitePCL.raw.sqlite3_column_blob(SQLitePCL.sqlite3_stmt,int)`

`  at Akavache.Sqlite3.SqliteOperationQueue.MarshalCompletion[T] (System.Object completion, System.Func`1[TResult] block, System.IObservable`1[T] commitResult) [0x00012] in D:\a\Akavache\Akavache\src\Akavache.Sqlite3\Queues\OperationQueue.cs:488 
--- End of stack trace from previous location where exception was thrown ---

  at System.Reactive.PlatformServices.ExceptionServicesImpl.Rethrow (System.Exception exception) [0x00000] in /_/Rx.NET/Source/src/System.Reactive/Internal/ExceptionServicesImpl.cs:19 
  at System.Reactive.ExceptionHelpers.Throw (System.Exception exception) [0x00000] in /_/Rx.NET/Source/src/System.Reactive/Internal/ExceptionServices.cs:16 
  at System.Reactive.Subjects.AsyncSubject`1[T].GetResult () [0x00038] in /_/Rx.NET/Source/src/System.Reactive/Subjects/AsyncSubject.cs:430 `

any solutions?

**Comment by mgramley** [comment]
> @mgramley did you find any solution? I am also facing the same issue while trying to use vS2022 . Akavache version 7.3.1 akavache.sqlite3 7.3.1

Sorry, I haven't found a solution, I've just reverted back to using 2019 for the time being.

**Comment by mgramley** [comment]
@protokovich it's nice to know I'm not the only one seeing this issue. I haven't found any solutions yet though.

**Comment by mgramley** [comment]
@glennawatson sorry it's taken me a bit, but I created a dirt-simple sample project that demonstrates the error.
https://github.com/mgramley/Akavache.MissingMethodExceptionSample

**Comment by Luiz-Franco** [comment]
If it helps, I found the same issue with VS 2022 preview and recently installed VS 2019 16.11.6 where the issue also happens.

The previous version I had installed was 16.10.4 and it did not have this problem.

**Comment by glennawatson** [comment]
Will have a look over the weekend and see what I can find with VS2022.

Won't be in a position to get to it sooner. I suspect VS2022 for some reason is loading a different SQLitePCL

Also probably worth lodging an issue in the SQLiteRaw repository https://github.com/ericsink/SQLitePCL.raw/issues

**Comment by glennawatson** [comment]
Might be worth adding a reference to the `SQLitePCL.Raw` nuget package as well in your platform projects (Android/iOS) and see if that helps.

**Comment by Luiz-Franco** [comment]
@glennawatson  I'll try that and get back to you.

And you should consider look into VS 2019 at least version 16.11.6 as well. As I stated above, the problem also happens on this version after I updated the IDE

**Comment by Luiz-Franco** [comment]
@glennawatson I should install the latest [SQLitePCLRaw.bundle_e_sqlite3](https://www.nuget.org/packages/SQLitePCLRaw.bundle_e_sqlite3) or a specific version?

**Comment by Luiz-Franco** [comment]
@glennawatson referencing [SQLitePCLRaw.bundle_e_sqlite3](https://www.nuget.org/packages/SQLitePCLRaw.bundle_e_sqlite3) did not help. Tried versions 2.0.4 and 2.0.7

**Comment by EmmanVazz** [comment]
Same issue. The only resolution I found is setting everything to BlobCache.InMemory.

**Comment by Luiz-Franco** [comment]
My workaround was to downgrade the VS 2019 version back to 16.10.4, last version that worked for me.

**Comment by glennawatson** [comment]
I've done a release that includes more up to date sqlite_lite_raw

Can anyone confirm if this helps with this problem?

Thanks

**Comment by protokovich** [comment]
@glennawatson it did not help

**Comment by glennawatson** [comment]
I've added an issue in @ericsink SQLitePCL.raw GitHub repo, noticing this issue is happening in multiple libraries now that rely on it with Xamarin. I suspect it may be new restrictions on P/Invokes introduced in newer Xamarin versions.

Also reached out to Eric directly.

**Comment by mgramley** [comment]
Has there been an update on this?

**Comment by ericsink** [comment]
@mgramley No update yet.  I'm still trying to figure this out.  Any progress I make will appear over at ericsink/SQLitePCL.raw#462

**Comment by glennawatson** [comment]
I'm keeping this issue open so people can quickly see it's being worked on but it's not a Akavache issue but the dependency SQLitePCLRaw, so please monitor in the link Eric provided.

**Comment by glennawatson** [comment]
Can everyone update to Visual Studio 2022 17.0.3  and see if it resolves the issue

**Comment by CreamyMiracle** [comment]
I'm facing this same exception but only when deploying Xamarin.iOS, not with Xamarin.Android

`System.MissingMethodException: 'Method not found: System.ReadOnlySpan1<byte> SQLitePCL.raw.sqlite3_column_blob(SQLitePCL.sqlite3_stmt,int)'`

Here is some background:

1. I was previously using VS2019 (16.11.8) and only developing Xamarin.Android - no issues.
2. Yesterday I decided to update VS2019 (16.11.8) to VS2022 (17.0.3) and for the first time ever I launched the same app on a physical iPhone device and faced this issue immediately. Android deployments are still working without issues with both VS2019 (16.11.8) and VS2022 (17.0.3).
3. I found this Github issue and re-installed VS2019 (16.11.8) to see if the problem would happen also with it when deploying Xamarin.iOS - yes it does

So I'm not sure if this is about VS2020 since I managed to reproduce it with VS2019 (16.11.8)
